backend/controllers/recommendationController.js

```diff
@@ -9,6 +9,22 @@ import { getPublicTemplates } from "./workoutTemplatesController.js";
 4. Sort the templates based on their final scores and return top results.
 */
 
+// Constants
+const weights = {
+  copyCount: 0.6,
+  netVotes: 0.25,
+  creatorFollowers: 0.15,
+};
+
+// Helper methods
+const normalizeValue = (value, max) => {
+  return max > 0 ? value / max : 0;
+};
+
+const normalizeVotes = (netVotes, maxAbsNetVotes) => {
+  return (netVotes + maxAbsNetVotes) / (2 * maxAbsNetVotes);
```

Comment:
Curious, why this equation instead of just `netVotes/maxAbsNetVotes`?

Reply:
This is because, netVotes could possibly be a negative number. Here's an example:

If `netVotes = -5` and the `maxAbsNetVotes = 15`, using the following formula `netVotes/maxAbsNetVotes` would result in the `normalizedVotes  =  -5/15 => -1/3`. So what I want to do is that I want to shift the votes range from `-15 to 15` to `0 to 30`. This way I don't have to deal with a negative normalizedVote. To do this, I make the numerator ` -5 + 15 = 10` and then the denominator `2 * 15 = 30`. This way the fraction changes from `-5/15` to `10/30` and the range changes from `-15 to 15` to `0 to 30`. Now the `normalizeVotes` cannot be a negative value. I hope I explained this well.

Reply:
what happens if `maxAbsNetVotes = -3?` Eg. no templates in the whole ecosystem have upvotes? We would still have negative values there right?

Reply:
maxAbsNetVotes is an absolute value so it cannot be -3. In this case, maxAbsNetVotes is positive 3. So per my formula: `(netVotes + maxAbsNetVotes) / (2 * maxAbsNetVotes)` => `(-3 + 3) / (2 * 6)` => `0/6` => `0`. This will be the correct value as -3 is the lowest possible value in the range -3 to 3 so in this case `normalizeVotes = 0`